build-logic-settings/configuration-cache-compatibility/src/main/kotlin/gradlebuild.configuration-cache-compatibility.settings.gradle.kts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import org.gradle.api.internal.plugins.DslObject
+
+run {
+
+    fun String.startsWithAnyOf(vararg prefixes: String): Boolean =
+        prefixes.any { prefix -> startsWith(prefix) }
+
+    fun Task.typeSimpleName(): String =
+        DslObject(this).declaredType.simpleName
+
+    val unsupportedTasksPredicate: (Task) -> Boolean = { task: Task ->
```

Comment:
❓ Why a `val` and not a `fun`?

Reply:
I moved it as it was in the `cc-experiment` file, initially trying to minimize the diff. But the extra indentation killed the effect.

Reply:
Since the PR got removed from the merge queue due to merge conflicts, I took the opportunity to change this to a function. Thanks for the suggestion!